### PR TITLE
Add clickForMoreInfo to statistics graph card

### PR DESCRIPTION
**Pull request by mib1185**
## Proposed change

This is adopts #19606 from history charts for statistics chart.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: #19144
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by karwosts** [comment]
I think this feature is problematic for mobile users (they can't activate tooltips anymore), and the original feature is undergoing some revision. Might make sense to wait for that to play out before expanding this wider.

**Comment by mib1185** [comment]
#19144 nice one 👍  i agree, we should wait for this

**Comment by karwosts** [comment]
Looks like we've gone ahead with making the more-info popups a mouse feature only, so if you wanted to replicate the same logic from https://github.com/home-assistant/frontend/pull/19606, that could probably go ahead now.

**Comment by mib1185** [comment]
changes from #19606 adopted

File: src/components/chart/statistics-chart.ts [hunk 1]
```diff
@@ -75,6 +76,8 @@ export class StatisticsChart extends LitElement {
 
   @property({ type: Boolean }) public isLoadingData = false;
 
+  @property({ type: Boolean }) public clickForMoreInfo = true;
```
Comment (karwosts) [inline src/components/chart/statistics-chart.ts:79]: Should set this to false when we're already in a more-info popup? 

See dialogs/more-info/ha-more-info-history.ts

**Comment by mib1185** [comment]
Hi @karwosts do you see a chance that we could get this in before beta cut?

**Comment by karwosts** [comment]
I don't really do merging, I let the maintainers handle that. But it looks like the first beta release has just gone out 😢

**Comment by bramkragten** [comment]
This will not be merged before we have consensus about the button logic for mobile, it could be merged if it would use the same logic as the current history does.

**Comment by mib1185** [comment]
it is the same, as it adopts #19606

File: src/components/chart/statistics-chart.ts [hunk 2]
```diff
@@ -245,6 +248,32 @@ export class StatisticsChart extends LitElement {
...
+          e,
+          "nearest",
+          { intersect: true },
+          true
+        );
+
+        if (points.length) {
+          const firstPoint = points[0];
+          fireEvent(this, "hass-more-info", {
+            entityId: this._statisticIds[firstPoint.datasetIndex],
```
Comment (bramkragten) [inline src/components/chart/statistics-chart.ts:272]: We should check if the statistics id is an entity id first

Comment (bramkragten) [reply src/components/chart/statistics-chart.ts:272]: Use `isExternalStatistic`:
https://github.com/home-assistant/frontend/blob/8f67ddf968eba518fb7d091bcc5cca6a37b0e98e/src/data/recorder.ts#L325

If it is an external statistic, we should not try to open the more info.

Comment (mib1185) [reply src/components/chart/statistics-chart.ts]: tbh, didn't know yet that there is something like "external statistics" ... so hopefully c8f9e32 covers what you mean?

File: src/components/chart/statistics-chart.ts [hunk 3]
```diff
@@ -245,6 +249,33 @@ export class StatisticsChart extends LitElement {
...
+        const points = chart.getElementsAtEventForMode(
+          e,
+          "nearest",
+          { intersect: true },
+          true
+        );
+
+        if (points.length) {
+          const firstPoint = points[0];
+          const enittyId = this._statisticIds[firstPoint.datasetIndex];
```
Comment (bramkragten) [inline src/components/chart/statistics-chart.ts:272]:
```suggestion
          const statisticId = this._statisticIds[firstPoint.datasetIndex];
```

File: src/components/chart/statistics-chart.ts [hunk 4]
```diff
@@ -245,6 +249,33 @@ export class StatisticsChart extends LitElement {
...
+          if (!isExternalStatistic(enittyId)) {
```
Comment (bramkragten) [inline src/components/chart/statistics-chart.ts:273]:
```suggestion
          if (!isExternalStatistic(statisticId)) {
```

File: src/components/chart/statistics-chart.ts [hunk 5]
```diff
@@ -245,6 +249,33 @@ export class StatisticsChart extends LitElement {
...
+            fireEvent(this, "hass-more-info", { entityId: enittyId });
```
Comment (bramkragten) [inline src/components/chart/statistics-chart.ts:274]:
```suggestion
            fireEvent(this, "hass-more-info", { entityId: statisticId });
```

**Review by bramkragten — APPROVED** [review]

**Comment by mib1185** [comment]
is it to late to get it in current beta? 😀

**Pull request merged**